ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -381,21 +426,21 @@ private static AttestationProcessingResult validateOnAttestation(
       return AttestationProcessingResult.INVALID;
     }
 
-    if (!store.getBlockRoots().contains(target.getRoot())) {
+    if (!forkChoiceStrategy.contains(target.getRoot())) {
       // Attestations target must be for a known block. If a target block is unknown, delay
       // consideration until the block is found
       return AttestationProcessingResult.UNKNOWN_BLOCK;
     }
 
-    if (!store.getBlockRoots().contains(attestation.getData().getBeacon_block_root())) {
+    if (!forkChoiceStrategy.contains(attestation.getData().getBeacon_block_root())) {
       // Attestations must be for a known block. If block is unknown, delay consideration until the
       // block is found
       return AttestationProcessingResult.UNKNOWN_BLOCK;
     }
 
-    if (store
-            .getBlock(attestation.getData().getBeacon_block_root())
-            .getSlot()
+    if (forkChoiceStrategy
+            .blockSlot(attestation.getData().getBeacon_block_root())
+            .get()
```

Comment:
If the previous checks that the roots are known mean we can safely call `get` here, can we replace one or more of them with a check that the returned `blockSlot` is empty?  Otherwise, do we need to handle blockSlot returning empty?

Reply:
No, you're right we don't need to handle `blockSlot` returning empty, we can remove the `.contains` check with just saving the return value from `forkChoiceStrategy.blockSlot`

Reply:
Done.